File: November 2023/Print Pattern/main.py
# Link: https://www.geeksforgeeks.org/problems/print-pattern3549/1

# User function Template for python3


class Solution:
    def pattern(self, N):
        temp = N
        if N <= 0:
            return [N]

        result = []
        while N > 0:
            result.append(N)
            N -= 5

        while N <= temp:
            result.append(N)
            N += 5

        return result


if __name__ == "__main__":
    N = 10
    ob = Solution()
    print(ob.pattern(N))

File: November 2023/Print Pattern/main.cpp
// Link: https://www.geeksforgeeks.org/problems/print-pattern3549/1

#include <bits/stdc++.h>
using namespace std;

// User function Template for C++

class Solution
{
public:
    vector<int> pattern(int N)
    {
        // code here
        vector<int> ans = {N};
        int n = N;
        while (N > 0)
        {
            ans.push_back(N -= 5);
        }
        while (N < n)
        {
            ans.push_back(N += 5);
        }
        return ans;
    }
};

int main()
{
    Solution obj;
    int N = 10;
    vector<int> ans = obj.pattern(N);
    for (int i = 0; i < ans.size(); i++)
    {
        cout << ans[i] << " ";
    }
    return 0;
}